fairscale/experimental/wgit/pygit.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import pathlib
+from pathlib import Path
+from typing import Union
+
+import pygit2
+
+
+class PyGit:
+    """
+    PyGit class to wrap the wgit/.git repo and interacta with the git repo.
+    """
+
+    def __init__(self, wgit_path: Union[str, pathlib.Path]) -> None:
+        # Find if a git repo exists within .wgit repo:
+        # If exists: then discover it and set the self.gitrepo path to its path
+        self.exists = None
+        self.wgit_path = wgit_path
+        self.path = Path(pygit2.discover_repository(self.wgit_path))
+
+        pygit_parent_p = self.path.parent.absolute()
+        if pygit_parent_p != self.wgit_path:
+            self.exists = False
+            self.repo = None
+            raise pygit2.GitError("No git repo exists within .wgit. Reinitialize weigit!")
+        else:
+            self.exists = True
+            self.repo = pygit2.Repository(str(self.path))
+            self.index = self.repo.index
+            self.name = "user"  # Commit metadata
+            self.email = "user@email.com"
```

Comment:
Yeah, I can see this is tricky. I think the init() needs to take those as optional arguments which can be used by test in case the user's config isn't set. For normal cases when user's .gitconfig is already set, the options won't be needed.